### DISCUSSION TITLE
UX: Program naming and potential searchability difficulty

### DISCUSSION THREAD

**Discussion by thomasrussellmurphy** [FiloSottile/age#239] [UX reports]
## What were you trying to do

Discover `age` as a potential tool to use having heard of it only by formal name or seen in a lightly-commented script file.

## What happened

Standalone, `age` is a simple English word and a notable substring of other words used in pack**age** man**age**rs. For example, while [Macports](https://www.macports.org/) is currently not a supported package manage for installing `age` on MacOS, the following searches show how the name is essentially useless for finding the package without necessarily an exact match:

```
$ port search --name age | grep Found
Found 313 ports.
$ port search age | grep Found
Found 2336 ports.
```

It would be fairly impractical to sort through these long lists trying to see if `age` is available without prior knowledge.

Search-confounding words include, but are not limited to: manager, management, language, package, damage, agent, image, message, usage.

Is there a way to improve the naming or searchability of the name of this utility?

**Comment by jasperweiss** [comment]
I was considering making a macports portfile for age but realised it would be impossible to find. This would have to be fixed first. I could name it "age-encryption" like the website, but it would probably help if the project itself was named as such.

**Comment by FiloSottile** [comment]
It's true that "age" alone is hard to search, but that's true of a lot of CLI binary names. We are currently the first Google result for "age command line tool" and "age encryption", so I think things worked out ok.
